be/src/pipeline/pipeline_x/operator.cpp

```diff
@@ -315,7 +315,7 @@ PipelineXLocalStateBase::PipelineXLocalStateBase(RuntimeState* state, OperatorXB
           _state(state),
           _finish_dependency(new FinishDependency(parent->operator_id(), parent->node_id(),
                                                   parent->get_name() + "_FINISH_DEPENDENCY")) {
-    _filter_dependency = std::make_unique<FilterDependency>(
+    _filter_dependency = std::make_shared<RuntimeFilterDependency>(
```

Comment:
`_filter_dependency` is only be hold in PipelineXLocalStateBase. So is it enough to use unique pointer?